mlrun/datastore/v3io.py

```diff
@@ -85,7 +85,12 @@ def get_filesystem(self, silent=True):
                     "v3iofs or storey not installed, run pip install storey"
                 ) from exc
             return None
-        self._filesystem = fsspec.filesystem("v3io", **self.get_storage_options())
```

Comment:
I would advise against altering the store.get_filesystem() API since it's an integral part of the DataStore base class. Instead, I would add a new user_store within the controller, incorporating the user's v3io_api_key for specific functionality.